src/Integration/State/TransferableVisualState.cs

```diff
@@ -29,14 +30,24 @@ namespace SonarLint.VisualStudio.Integration.State
     internal class TransferableVisualState : ViewModelBase
     {
         private readonly ObservableCollection<ServerViewModel> connectedServers = new ObservableCollection<ServerViewModel>();
+        private readonly IThreadHandling threadHandling;
         private ProjectViewModel boundProject;
         private bool isBusy;
 
+        public TransferableVisualState()
+            : this(ThreadHandling.Instance)
+        { }
+
+        internal /* for testing */ TransferableVisualState(IThreadHandling threadHandling)
+        {
+            this.threadHandling = threadHandling;
+        }
+
         public ObservableCollection<ServerViewModel> ConnectedServers
         {
             get
             {
-                Debug.Assert(ThreadHelper.CheckAccess(), $"{nameof(ConnectedServers)} should only be accessed from the UI thread");
```

Comment:
This class had lots of assertions using `ThreadHelper`. I've replaced them with `IThreadHandling`, but to do that I had to add some constructors so the tests can supply the component to use.